stellargraph/layer/graph_classification.py

```diff
@@ -0,0 +1,255 @@
+
+      - There are 2 inputs required, the node features and the normalized graph adjacency matrices for each batch entry.
+
+      - This class assumes that the normalized graph adjacency matrices are passed as input to the Keras methods.
+
+    Args:
+        units (int): dimensionality of output feature vectors
+        activation (str): nonlinear activation applied to layer's output to obtain output features
+        use_bias (bool): toggles an optional bias
+
+        kernel_initializer (str or func, optional): The initialiser to use for the weights of each graph
+            convolutional layer.
+        kernel_regularizer (str or func, optional): The regulariser to use for the weights of each graph
+            convolutional layer.
+        kernel_constraint (str or func, optional): The constraint to use for the weights of each layer graph
+            convolutional.
+        bias_initializer (str or func, optional): The initialiser to use for the bias of each layer graph
+            convolutional.
+        bias_regularizer (str or func, optional): The regulariser to use for the bias of each layer graph
+            convolutional.
+        bias_constraint (str or func, optional): The constraint to use for the bias of each layer graph
+            convolutional.
+     """
+
+    def call(self, inputs):
+        """
+        Applies the layer.
+
+        Args:
+            inputs (list or tuple): a list or tuple of 2 input tensors that includes
+                node features (size batch_size x N x F),
+                graph adjacency matrix (batch_size x N x N),
+                where N is the number of nodes in the graph, and F is the dimensionality of node features.
+
+        Returns:
+            Keras Tensor that represents the output of the layer.
+        """
+        features, A = inputs
+
+        h_graph = K.batch_dot(A, features)
+        output = K.dot(h_graph, self.kernel)
+
+        # Add optional bias & apply activation
+        if self.bias is not None:
+            output += self.bias
+        output = self.activation(output)
+
+        return output
+
+
+@experimental(reason="Missing unit tests")
+class GCNSupervisedGraphClassification:
+    """
+    A stack of :class:`GraphClassificationConvolution` layers together with a Keras `GlobalAveragePooling1D` layer
+    that implement a supervised graph classification network using the GCN convolution operator
+    (https://arxiv.org/abs/1609.02907).
+
+    The model minimally requires specification of the GCN layer sizes as a list of ints
+    corresponding to the feature dimensions for each hidden layer,
+    activation functions for each hidden layers, and a generator object.
+
+    To use this class as a Keras model, the features and pre-processed adjacency matrix
+    should be supplied using the :class:`GraphGenerator` class.
+
+    Notes:
+      - The inputs are tensors provided by the :class:`GraphGenerator` object.
```

Comment:
I think this note and the paragraph immediately before it are saying essentially the same thing. Maybe they could be combined?